sdk/kiosk/src/client/kiosk-transaction.ts

```diff
@@ -0,0 +1,461 @@
+			share: true,
+			transfer: true,
+		});
+		const [kiosk, cap] = kioskTx.createKiosk(this.txb);
+		this.kiosk = kiosk;
+		this.kioskCap = cap;
+		return this;
+	}
+
+	/**
+	 * Creates a personal kiosk & shares it.
+	 * The `PersonalKioskCap` is transferred to the signer.
+	 * @param borrow If true, the `kioskOwnerCap` is borrowed from the `PersonalKioskCap` to be used in next transactions.
+	 */
+	createPersonal(borrow?: boolean) {
+		this.#pendingShare = true;
+		return this.create().convertToPersonal(borrow);
+	}
+
+	/**
+	 * Converts a kiosk to a Personal (Soulbound) Kiosk.
+	 * Requires initialization by either calling `ktxb.create()` or `ktxb.setCap()`.
+	 */
+	convertToPersonal(borrow?: boolean) {
+		this.#validateKioskIsSet();
+
+		const cap = convertToPersonalTx(
+			this.txb,
+			this.kiosk!,
+			this.kioskCap!,
+			this.kioskClient.getRulePackageId('personalKioskRulePackageId'),
+		);
+
+		// if we enable `borrow`, we borrow the kioskCap from the cap.
+		if (borrow) this.#borrowFromPersonalCap(cap);
+		else this.#personalCap = cap;
+
+		this.#setPendingStatuses({ transfer: true });
+		return this;
+	}
+
+	/**
+	 * Single function way to create a kiosk, share it and transfer the cap to the specified address.
+	 */
+	createAndShare(address: string) {
+		const cap = kioskTx.createKioskAndShare(this.txb);
+		this.txb.transferObjects([cap], this.txb.pure(address, 'address'));
+	}
+
+	/**
+	 * Shares the kiosk.
+	 */
+	share() {
+		this.#validateKioskIsSet();
+		this.#setPendingStatuses({ share: false });
+		kioskTx.shareKiosk(this.txb, this.kiosk!);
+	}
+
+	/**
+	 * Should be called only after `create` is called.
+	 * It shares the kiosk & transfers the cap to the specified address.
+	 */
+	shareAndTransferCap(address: string) {
+		if (this.#personalCap)
+			throw new Error('You can only call `shareAndTransferCap` on a non-personal kiosk.');
+		this.#setPendingStatuses({ transfer: false });
+		this.share();
+		this.txb.transferObjects([this.kioskCap!], this.txb.pure(address, 'address'));
+	}
+
+	/**
+	 * A function to borrow an item from a kiosk & execute any function with it.
+	 * Example: You could borrow a Fren out of a kiosk, attach an accessory (or mix), and return it.
+	 */
+	borrowTx({ itemType, itemId }: ItemId, callback: (item: TransactionArgument) => void) {
+		this.#validateKioskIsSet();
+		const [itemObj, promise] = kioskTx.borrowValue(
+			this.txb,
+			itemType,
+			this.kiosk!,
+			this.kioskCap!,
+			itemId,
+		);
+
+		callback(itemObj);
+
+		this.return({ itemType, item: itemObj, promise });
+	}
+
+	/**
+	 * Borrows an item from the kiosk.
+	 * This will fail if the item is listed for sale.
+	 *
+	 * Requires calling `return`.
+	 */
+	borrow({ itemType, itemId }: ItemId): [TransactionArgument, TransactionArgument] {
+		this.#validateKioskIsSet();
+		const [itemObj, promise] = kioskTx.borrowValue(
+			this.txb,
+			itemType,
+			this.kiosk!,
+			this.kioskCap!,
+			itemId,
+		);
+
+		return [itemObj, promise];
```

Comment:
I might make these kinds of returns objects rather than tuples, but I don't feel too strongly about it